ginkgo-kubernetes-all.Jenkinsfile

```diff
@@ -85,70 +125,122 @@ pipeline {
             options {
                 timeout(time: 100, unit: 'MINUTES')
             }
-            steps {
-                script {
-                    parallel(
-                        "K8s-1.11":{
-                            sh 'cd ${TESTDIR}; K8S_VERSION=1.11 ginkgo --focus=" K8s*" -v --failFast=${FAILFAST}'
-                        },
-                        "K8s-1.12":{
-                            sh 'cd ${TESTDIR}; K8S_VERSION=1.12 ginkgo --focus=" K8s*" -v --failFast=${FAILFAST}'
-                        },
-                        failFast: "${FAILFAST}".toBoolean()
-                    )
+            failFast true
+            parallel {
+                stage('BDD-Test-k8s-1.11') {
+                    environment {
+                        TESTED_SUITE="1.11"
+                        GOPATH="${WORKSPACE}/${TESTED_SUITE}-gopath"
+                        TESTDIR="${GOPATH}/${PROJ_PATH}/test"
+                    }
+                    steps {
+                        sh 'cd ${TESTDIR}; K8S_VERSION=${TESTED_SUITE} ginkgo --focus=" K8s*" -v --failFast=${FAILFAST} -- -cilium.provision=false'
+                    }
+                    post {
+                        always {
+                            sh 'cd ${TESTDIR}; ./post_build_agent.sh || true'
+                            sh 'cd ${TESTDIR}; ./archive_test_results.sh || true'
```

Comment:
I just realised, does archive_test_results.sh race between stages? It does a `cd ${WORKSPACE}/test` and copies files one layer up, and we then move them on line 143 below. What happens if multiple parallel test runs complete at the same time? The `mv` and `rm` calls might collide? (this might already be true with the similar runtime changes)

Reply:
`${TESTDIR}` is separate for each test run. 

`TESTDIR` here expands to ${WORKSPACE}/${TESTED_SUITE}-gopath/src/github.com/cilium/cilium/test

${TESTED_SUITE} is different for each stage.